torch/testing/_internal/common_modules.py

```diff
@@ -1399,6 +1399,15 @@ def module_inputs_torch_nn_LSTM(module_info, device, dtype, requires_grad, train
                skips=(
                    DecorateInfo(skipIfMps, 'TestModule', dtypes=[torch.float64]),),
                supports_gradgrad=False),
+    # TransformerEncoder takes the same inputs as TransformerEncoderLayer
+    ModuleInfo(torch.nn.TransformerEncoder,
+               train_and_eval_differ=True,
```

Comment:
In which case does training and evaluation differs?

Reply:
It calls TransformerEncoderLayer which differs according to pre-existing ModuleInfo entry, so I transitively bubbled up the flag.